tests/inputFieldsSpec.js

```diff
@@ -521,5 +521,33 @@
             });
 
         });
+
+        describe('Number padding filter', function () {
+            var numberPaddingFilter;
+
+            beforeEach(function () {
+                inject(function (_numberPaddingFilter_) {
+                    numberPaddingFilter = _numberPaddingFilter_;
+                });
+            });
+
+            it('should add leading 0s to single digits', function () {
+                expect(numberPaddingFilter(0, '2')).toBe('00');
+            });
+
+            it('should not add leading 0s to double digits', function () {
+                expect(numberPaddingFilter(11, '2')).toBe('11');
+            });
+
+            it('should not add leading 0s if second argument is 0', function () {
+                expect(numberPaddingFilter(0, '0')).toBe('0');
+            });
+
+            it('should not trim the double digit number if second argument is 0', function () {
+                expect(numberPaddingFilter(10, '0')).toBe('10');
+            });
```

Comment:
Why are the second parameters strings? Add some tests using numbers, also different lengths to `0` and `2`.